common/src/test/java/io/netty/util/concurrent/DefaultPromiseTest.java

```diff
@@ -143,6 +143,38 @@ public void testListenerNotifyLater() throws Exception {
         testListenerNotifyLater(2);
     }
 
+    @Test(timeout = 2000)
+    public void promiseListenerAddWhenCompleteFailure() throws Exception {
+        promiseListenerAddWhenComplete(new RuntimeException());
+    }
+
+    @Test(timeout = 2000)
+    public void promiseListenerAddWhenCompleteSuccess() throws Exception {
+        promiseListenerAddWhenComplete(null);
+    }
+
+    private static void promiseListenerAddWhenComplete(Throwable cause) throws InterruptedException {
```

Comment:
just a nit but do we want to also prefix the new methods with "test" as the others in the class ?